src/xnnpack/intrinsics-polyfill.h

```diff
@@ -317,4 +330,40 @@ HVX_Vector Q6_Vw_vmpyie_VwVh(HVX_Vector multiplier_lo, HVX_Vector multiplier_hi,
 
     return vout;
 }
-#endif  // Hexagon
+
+// Horizontal vector sum by pairwise addition.
+// To calculate fewer elements than the full 128 bytes in 'vin', 
+// use the following code first before calling the intrinsic:
+//   vin = Q6_V_vand_QV(Q6_Q_vsetq_R(batch), vin);
+// where 'batch' is equal to 'elements * sizeof(float)'
+static XNN_INTRINSIC
+float Q6_f32_vrsum_Vsf(HVX_Vector vin){
+    HVX_VectorPair vsum_pair = Q6_W_vshuff_VVR(vin, vin, 64);
+    vin = Q6_Vsf_vadd_VsfVsf(Q6_V_lo_W(vsum_pair), Q6_V_hi_W(vsum_pair));
+
+    vsum_pair = Q6_W_vshuff_VVR(vin, vin, 32);
+    vin = Q6_Vsf_vadd_VsfVsf(Q6_V_lo_W(vsum_pair), Q6_V_hi_W(vsum_pair));
+
+    vsum_pair = Q6_W_vshuff_VVR(vin, vin, 16);
+    vin = Q6_Vsf_vadd_VsfVsf(Q6_V_lo_W(vsum_pair), Q6_V_hi_W(vsum_pair));
+
+    vsum_pair = Q6_W_vshuff_VVR(vin, vin, 8);
+    vin = Q6_Vsf_vadd_VsfVsf(Q6_V_lo_W(vsum_pair), Q6_V_hi_W(vsum_pair));
+
+    vsum_pair = Q6_W_vshuff_VVR(vin, vin, 4);
+    vin = Q6_Vsf_vadd_VsfVsf(Q6_V_lo_W(vsum_pair), Q6_V_hi_W(vsum_pair));
+
+    return *((float *) &vin);
+}
+
+static XNN_INTRINSIC
+HVX_Vector Q6_Vsf_vdiv_VsfVsf(HVX_Vector vin1, HVX_Vector vin2){
+    float* svin1 = (float *) &vin1;
+    float* svin2 = (float *) &vin2;
+
+    for(int i = 0; i < 32; i++)
+      svin1[i] = svin1[i] / svin2[i];
```

Comment:
suggest adding a todo to improve on this.   if div is not an option, perhaps try to make calling code use 'nr' - reciprocal.

Reply:
Quick question - does XNNPACK has nr code already implemented or should we add? I added as a TODO for now, but definitely want to improve this.